ttnn/cpp/ttnn/operations/eltwise/binary_ng/binary_ng.cpp

```diff
@@ -16,8 +16,17 @@ Tensor BinaryNg<binary_op_type>::invoke(
     const std::optional<const DataType> &output_dtype,
     const std::optional<MemoryConfig> &memory_config,
     std::optional<Tensor> optional_output_tensor) {
+    Tensor input_a = input_tensor_a;
+    if (input_tensor_a.get_dtype() != DataType::BFLOAT16) {
+        input_a = ttnn::typecast(input_tensor_a, DataType::BFLOAT16);
+    }
+    Tensor input_b = input_tensor_b;
+    if (input_tensor_b.get_dtype() != DataType::BFLOAT16) {
+        input_b = ttnn::typecast(input_tensor_b, DataType::BFLOAT16);
+    }
+
     return ttnn::prim::binary_ng(
-        queue_id, input_tensor_a, input_tensor_b, binary_op_type, output_dtype, memory_config, optional_output_tensor);
+        queue_id, input_a, input_b, binary_op_type, output_dtype, memory_config, optional_output_tensor);
```

Comment:
Why not define a function for this since it's used a few times?
```cpp
inline Tensor typecast_to(DataType dtype, const Tensor &input) {
  return input.get_dtype() == dtype ? input : ttnn::typecast(input, dtype);
}
```
then
```suggestion
    Tensor input_a = typecast_to(DataType::BFLOAT16, input_tensor_a);
    Tensor input_b = typecast_to(DataType::BFLOAT16, input_tensor_b);

    return ttnn::prim::binary_ng(
        queue_id, input_a, input_b, binary_op_type, output_dtype, memory_config, optional_output_tensor);
```

Reply:
Updated